src/crypto/crypto_sig.cc

```diff
@@ -93,11 +93,12 @@ std::unique_ptr<BackingStore> Node_SignFinal(Environment* env,
   }
   EVPKeyCtxPointer pkctx(EVP_PKEY_CTX_new(pkey.get(), nullptr));
   if (pkctx &&
-      EVP_PKEY_sign_init(pkctx.get()) &&
+      EVP_PKEY_sign_init(pkctx.get()) > 0 &&
       ApplyRSAOptions(pkey, pkctx.get(), padding, pss_salt_len) &&
-      EVP_PKEY_CTX_set_signature_md(pkctx.get(), EVP_MD_CTX_md(mdctx.get())) &&
+      EVP_PKEY_CTX_set_signature_md(pkctx.get(),
+                                    EVP_MD_CTX_md(mdctx.get())) > 0 &&
       EVP_PKEY_sign(pkctx.get(), static_cast<unsigned char*>(sig->Data()),
-                    &sig_len, m, m_len)) {
+                    &sig_len, m, m_len) > 0) {
```

Comment:
Previously when `EVP_PKEY_sign` and `EVP_PKEY_sign_init` return `-2`, we let it slip into the successful branch because `-2` is also cast into `true`. I suggest we update all OpenSSL API usages from `if(EVP_FUNC())` to `if(EVP_FUNC() > 0)`, so that if the API starts to return negative values, we will not accidentally jump into the successful branch.